File: FanTastyBack/Models/Utilisateur.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;

namespace FanTastyBack.Models
{
    public class Utilisateur
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Nom { get; set; }

        public string Prenom { get; set; }  

        public string Email { get; set; }   

        public string MotDePasse { get; set; }  

        [BsonRepresentation(BsonType.ObjectId)]
        public List<string> RecetteFavoris { get; set; } = new List<string>();

        [BsonRepresentation(BsonType.ObjectId)]
        public List<string> PreferenceIngredients { get; set; }

        public Utilisateur(string id, string nom, string prenom, string email)
        {
            Id = id;
            Nom = nom;
            Prenom = prenom;
            Email = email;

        }

    }
}

File: FanTastyFront/src/app/models/utilisateur.ts
export class Utilisateur {
  public id : string | null;
  public nom: string;
  public prenom: string;
  public email: string;
  public recetteFavoris: string[];
  public preferenceIngredients: string[];
  public roles: string[];
  private _token!: string | null;
  private _password!: string | null;

  /**
   *
   * @param id
   * @param nom
   * @param prenom
   * @param email
   * @param recetteFavoris
   * @param preferenceIngredients
   * @param roles
   */
  constructor(id: string | null, nom: string, prenom: string, email: string, recetteFavoris: string[], preferenceIngredients: string[], roles: string[]) {
    this.id = id;
    this.nom = nom;
    this.prenom = prenom;
    this.email = email;
    this.recetteFavoris = recetteFavoris;
    this.preferenceIngredients = preferenceIngredients;
    this.roles = roles;
  }

  public set token(token: string) {
    this._token = token;
  }

  public get token() {
    return this._token!;
  }

  public set password(password: string) {
    this._password = password;
  }

  public get password() {
    return this._password!;
  }
}
